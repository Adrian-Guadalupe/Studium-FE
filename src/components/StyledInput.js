import React from "react";
import styled from "styled-components";
import { Input, Form } from "antd";
import PropTypes from "prop-types";

const StyledInput = ( { bordered = false, borderRadius, label, ...props } ) => {
  if( bordered ){
    return ( <StyledFormItem label={ label }>
      <StyledAntdInput { ...props } borderRadius={ borderRadius }/>
    </StyledFormItem> );
  }else{
    return ( <StyledFormItem label={ label }>
      <StyledBorderBottom>
        <StyledNoBorderAntdInput { ...props }>
        
        </StyledNoBorderAntdInput>
      </StyledBorderBottom>
    </StyledFormItem> );
  }
};
const StyledFormItem = styled( Form.Item )`
  && {
    text-align: left;
    width: 90%;
  }
`;

const StyledBorderBottom = styled.div`
  border-bottom: 1px solid gray;
`;

const StyledNoBorderAntdInput = styled( Input )`
  &&{
    border: 0px;
    :focus {
      box-shadow: none;
    }
  }
`;

const StyledAntdInput = styled( Input )`
  &&{
    border-radius: ${ props => props.borderRadius ? props.borderRadius + "px" :
  +"6px" };
    :focus {
      box-shadow: none;
    }
  }
`;

StyledInput.propTypes = {
  onChange: PropTypes.func.isRequired,
  bordered: PropTypes.bool,
  value: PropTypes.any.isRequired,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  size: PropTypes.oneOf( [ "large", "default", "small" ] ),
  borderRadius: PropTypes.number
  
};

export default StyledInput;