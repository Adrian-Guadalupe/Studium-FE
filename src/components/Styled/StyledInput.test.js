import React from 'react';
import renderer from 'react-test-renderer';

import StyledInput from './StyledInput';

test('StyledInput render correctly', () => {
    const tree = renderer.create(<StyledInput />).toJSON();
    expect(tree).toMatchSnapshot();
});