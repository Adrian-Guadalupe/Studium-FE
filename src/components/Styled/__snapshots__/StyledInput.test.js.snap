// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledInput render correctly 1`] = `
.c0.c0 {
  text-align: left;
  width: 90%;
}

.c1.c1:focus {
  box-shadow: none;
}

<div
  className="ant-row ant-form-item c0"
  style={Object {}}
>
  <div
    className="ant-col ant-form-item-control-wrapper"
    style={Object {}}
  >
    <div
      className="ant-form-item-control"
    >
      <span
        className="ant-form-item-children"
      >
        <input
          className="ant-input c1"
          onChange={[Function]}
          onKeyDown={[Function]}
          type="text"
          value=""
        />
      </span>
    </div>
  </div>
</div>
`;
